playground/cdk/CdkSample.AppHost/storage.module.bicep

```diff
@@ -1,77 +1,69 @@
-    principalId: principalId
-    principalType: principalType
-  }
+resource StorageBlobDataContributor_storage 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'StorageBlobDataContributor_storage')
+    properties: {
+        principalId: principalId
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')
+        principalType: principalType
+    }
+    scope: storage
 }
 
-resource roleAssignment_HRj6MDafS 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
-  scope: storageAccount_1XR3Um8QY
-  name: guid(storageAccount_1XR3Um8QY.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3'))
-  properties: {
-    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3')
-    principalId: principalId
-    principalType: principalType
-  }
+resource StorageTableDataContributor_storage 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'StorageTableDataContributor_storage')
+    properties: {
+        principalId: principalId
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3')
+        principalType: principalType
+    }
+    scope: storage
 }
 
-resource roleAssignment_r0wA6OpKE 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
-  scope: storageAccount_1XR3Um8QY
-  name: guid(storageAccount_1XR3Um8QY.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '974c5e8b-45b9-4653-ba55-5f855dd0fb88'))
-  properties: {
-    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '974c5e8b-45b9-4653-ba55-5f855dd0fb88')
-    principalId: principalId
-    principalType: principalType
-  }
+resource StorageQueueDataContributor_storage 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'StorageQueueDataContributor_storage')
+    properties: {
+        principalId: principalId
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '974c5e8b-45b9-4653-ba55-5f855dd0fb88')
+        principalType: principalType
+    }
+    scope: storage
 }
 
-output blobEndpoint string = storageAccount_1XR3Um8QY.properties.primaryEndpoints.blob
-output queueEndpoint string = storageAccount_1XR3Um8QY.properties.primaryEndpoints.queue
-output tableEndpoint string = storageAccount_1XR3Um8QY.properties.primaryEndpoints.table
+output blobEndpoint string = storage.properties.primaryEndpoints.blob
+
+output queueEndpoint string = storage.properties.primaryEndpoints.queue
+
+output tableEndpoint string = storage.properties.primaryEndpoints.table
+
+param storagesku string
+
+param locationOverride string
+
+param principalId string
+
+param principalType string
```

Comment:
Can you fix it so these are first? This is so hard to look at 😄

Reply:
yeah, it's a pain to reorder everything manually every time I want to diff the before/after.

@tg-msft - is the order in the file based on the order in the collection in memory? I can try to reorder things to make all parameters first.

Reply:
We want to.

Reply:
This is now done. The parameters aren't always exactly in the same order, but all params are first and all outputs are last. It makes the diff a lot easier.

Reply:
I added back part of module splitting to sort these in https://github.com/Azure/azure-sdk-for-net/pull/45911.